pkg/statistics/handle/autoanalyze/autoanalyze.go

```diff
@@ -76,6 +78,102 @@ func (sa *statsAnalyze) DeleteAnalyzeJobs(updateTime time.Time) error {
 	})
 }
 
+// CleanupCorruptedAnalyzeJobs cleans up the potentially corrupted analyze job.
+// It cleans up both the job from current owner and the job from other nodes.
+func (sa *statsAnalyze) CleanupCorruptedAnalyzeJobs(currentRunningProcessIDs map[uint64]struct{}) error {
+	return statsutil.CallWithSCtx(sa.statsHandle.SPool(), func(sctx sessionctx.Context) error {
+		return CleanupCorruptedAnalyzeJobs(sctx, currentRunningProcessIDs)
+	})
+}
+
+// SelectAnalyzeJobsSQL is the SQL to select the analyze jobs whose
+// state is `pending` or `running` and the update time is more than 10 minutes ago.
+const SelectAnalyzeJobsSQL = `SELECT id, process_id, instance
+		FROM mysql.analyze_jobs
+		AND state IN ('pending', 'running')
+		AND TIMESTAMPDIFF(MINUTE, update_time, NOW()) > 10`
+
+// BatchUpdateAnalyzeJobSQL is the SQL to update the analyze jobs to `failed` state.
+const BatchUpdateAnalyzeJobSQL = `UPDATE mysql.analyze_jobs
+            SET state = 'failed',
+            fail_reason = 'TiDB Server is down when running the analyze job',
+            process_id = NULL
+            WHERE id IN (%?)`
```

Comment:
If too much data need to be updated. it will happen the problem.

Reply:
Can you provide more information about the problem and the data threshold that triggers it? We typically don't have too many tasks running simultaneously.

Reply:
We have some users with numerous tables who will write their own scripts to execute analyze in a distributed manner, which may result in a large number of analyze jobs.

Reply:
But I don't think they will have thousands of tasks on the same node at the same time.

Reply:
After our offline discussion, we decided to keep it because we usually don't have that many tasks at the same time.